fastestimator/xai/saliency.py

```diff
@@ -136,7 +135,7 @@ def _get_mask(self, batch: Dict[str, Any]) -> Dict[str, Tensor]:
         for key in self.gather_keys:
             # If there's no target key, use an empty array which will cause the max-likelihood class to be selected
             batch.setdefault(key, [])
-        batch, prediction = self.network.run_step(batch=batch)
+        prediction = self.network.transform(data=batch, mode=self.mode)
```

Comment:
why? transform loads and unloads every time right? This method is called repeatedly to generate smoothed masks so this will waste significant time.

Reply:
there's a multi-gpu inferencing problem.   the fix is in `transform`, it's not ideal to apply the fix in separate places.

also, `transform` is exactly designed for executing network operations in non-tensor places like Traces.

Reply:
transform calls load_epoch (with warmup=True) every time it is invoked. XAI loops this method about 50 times by default during it's execution. That's a lot of overhead

Reply:
took 0.00024980400121421553 seconds
 took 0.00022457700106315315 seconds
 took 0.00021970600027998444 seconds
 took 0.00023631700059922878 seconds
 took 0.00026834100026462693 seconds
 took 0.0002539209999667946 seconds
 took 0.0002640370003064163 seconds
 took 0.00024317899988091085 seconds
 took 0.00021316400125215296 seconds
 took 0.0002047990001301514 seconds
 took 0.00023421199875883758 seconds
 took 0.00023299299937207252 seconds